packages/transaction-controller/src/utils/batch.ts

```diff
@@ -106,24 +106,26 @@ export const ERROR_MESSAGE_NO_UPGRADE_CONTRACT =
 export async function addTransactionBatch(
   request: AddTransactionBatchRequest,
 ): Promise<TransactionBatchResult> {
-  const { getInternalAccounts, messenger, request: userRequest } = request;
+  const {
+    getInternalAccounts,
+    messenger,
+    request: transactionBatchRequest,
+  } = request;
   const sizeLimit = getBatchSizeLimit(messenger);
 
   validateBatchRequest({
     internalAccounts: getInternalAccounts(),
-    request: userRequest,
+    request: transactionBatchRequest,
     sizeLimit,
   });
 
-  const { useHook } = userRequest;
+  log('Adding', transactionBatchRequest);
 
-  log('Adding', userRequest);
-
-  if (useHook) {
-    return await addTransactionBatchWithHook(request);
+  if (!transactionBatchRequest.disable7702) {
+    return await addTransactionBatchWith7702(request);
```

Comment:
Does this not mean that if 7702 is enabled, but the chain isn't supported, it will throw and not fallback to anything else?

Do we need to use a try catch and only handle the specific chain not supported error?